api.js

```diff
@@ -1044,6 +1140,52 @@
         params.token = this.token;
       }
 
+      // Let's see if we should cache this route
+      var cancelRequest = false,
+          hash          = null,
+          cacheAction   = null;
+      cachedRoutes.forEach(function(action) {
+
+        // Some other route must've matched
+        if (cancelRequest) return;
+
+        // Some filters to cancel caching
+        if (action.method.toLowerCase()!=lcMethod) return;
+
+        // Test if the action wants the current url
+        var urlParts = action.route.exec(url);
+        if (!urlParts) return;
+
+        // Generate hash source (and last option to non-caching behavior)
+        var hashSource = action.generateHash(urlParts, params);
+        if (!hashSource) return;
+        hashSource = [action.route.source, hashSource];
+
+        // Store action & actually hash the damn thing
+        cacheAction = action;
+        hash = 'ClubIsLiveApiClient-' + qhash(hashSource);
+
+        // Check if we have data
+        var data = window.localStorage.getItem(hash);
+        if (data) {
+          try {
+            data = JSON.parse(data);
+          } catch(e) {
+            console.warn('Invalid cache data for', action.route);
+            return;
+          }
+          // Return or expire
+          if (data.expires > (new Date()).getTime()) {
+            callback(data.error, data.response);
+            cancelRequest = true;
+          } else {
+            window.localStorage.removeItem(hash);
+          }
+        }
+      });
+      // We've used cache & called the callback already. Let's stop now
+      if (cancelRequest) return;
```

Comment:
there is only one place where this is set to true, so maybe you can return there immediately